src/app/AttributeAccessInterfaceRegistry.h

```diff
@@ -16,39 +16,50 @@
 #pragma once
 
 #include <app/AttributeAccessInterface.h>
+#include <app/AttributeAccessInterfaceCache.h>
 #include <app/util/af-types.h>
 
-/**
- * Register an attribute access override.  It will remain registered until the
- * endpoint it's registered for is disabled (or until shutdown if it's
- * registered for all endpoints) or until it is explicitly unregistered.
- * Registration will fail if there is an already-registered override for the
- * same set of attributes.
- *
- * @return false if there is an existing override that the new one would
- *               conflict with.  In this case the override is not registered.
- * @return true if registration was successful.
- */
-bool registerAttributeAccessOverride(chip::app::AttributeAccessInterface * attrOverride);
+namespace chip {
+namespace app {
 
-/**
- * Unregister an attribute access override (for example if the object
- * implementing AttributeAccessInterface is being destroyed).
- */
-void unregisterAttributeAccessOverride(chip::app::AttributeAccessInterface * attrOverride);
+class AttributeAccessInterfaceRegistry
+{
+public:
+    /**
+     * Register an attribute access override.  It will remain registered until the
+     * endpoint it's registered for is disabled (or until shutdown if it's
+     * registered for all endpoints) or until it is explicitly unregistered.
+     * Registration will fail if there is an already-registered override for the
+     * same set of attributes.
+     *
+     * @return false if there is an existing override that the new one would
+     *               conflict with.  In this case the override is not registered.
+     * @return true if registration was successful.
+     */
+    bool RegisterAttributeAccessOverride(chip::app::AttributeAccessInterface * attrOverride);
```

Comment:
I know you kept the existing naming, but that naming was for a context-less free function.  In the context of this class, should this be perhaps called `Register`?  Or `RegisterAttributeAccess` or something?

Reply:
Yes, Started renaming and it started lowercase with `register` being a reserved word. Updated.